homeassistant/components/iron_os/entity.py

```diff
@@ -0,0 +1,45 @@
+"""Base entity for Pinecil integration."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from homeassistant.helpers.device_registry import CONNECTION_BLUETOOTH, DeviceInfo
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import MANUFACTURER, MODEL
+from .coordinator import PinecilCoordinator
+
+
+class PinecilBaseEntity(CoordinatorEntity[PinecilCoordinator]):
+    """Base Pinecil entity."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: PinecilCoordinator,
+        entity_description: EntityDescription,
+        context: Any = None,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        if TYPE_CHECKING:
+            assert coordinator.device
+            assert coordinator.config_entry
```

Comment:
Overwrite the config_entry type in the coordinator.
I would maybe suggest to do the same for device, but I think there is a reason you dont have this yet